pennylane/measurements/probs.py

```diff
@@ -139,30 +139,20 @@ def return_type(self):
     def numeric_type(self):
         return float
 
-    def shape(self, device=None):
-        if qml.active_return():
-            return self._shape_new(device)
-        if device is None:
-            raise MeasurementShapeError(
-                "The device argument is required to obtain the shape of the measurement "
-                f"{self.__class__.__name__}."
-            )
+    def _shape_legacy(self, device, shots):
         num_shot_elements = (
-            1 if device.shot_vector is None else sum(s.copies for s in device.shot_vector)
+            1 if len(shots.shot_vector) <= 1 else sum(s.copies for s in shots.shot_vector)
```

Comment:
```suggestion
            sum(s.copies for s in shots.shot_vector) if shots.has_partitioned_shots else 1
```
We must consider the case `Shots((1,1))`.  The length of the shot vector is only 1, but there's two copies.  


Take for example:
```
>>> dev = qml.device('default.qubit', wires=2, shots=(1,2))
>>> qs = qml.tape.QuantumScript([], [qml.probs(wires=(0,1))])
>>> qs.shape(dev)
(4,)
```
It should be instead `(4,4)`.